### ISSUE TITLE
deserializeSync/fromJSON is mutating the input parameter

### ISSUE THREAD

**Issue by gajus** [salesforce/tough-cookie#86]
This took me quite some time to trace down...

I was doing:

```js
console.log(CookieJar.deserializeSync(scrapeSeatsSession.jar));

const seats = await getEventSeats(
  CookieJar.fromJSON(scrapeSeatsSession.jar),
  scrapeSeatsSession.sessionId
);
```

and I couldn't understand where my cookies disappear. It turns out that the first call to `CookieJar.deserializeSync(scrapeSeatsSession.jar)` drains the `scrapeSeatsSession.jar`.

As a very least, this should be documented with a big red flag.

**Comment by gajus** [comment]
Meanwhile, a workaround is to deep clone the object, e.g. `CookieJar.fromJSON(JSON.parse(JSON.stringify(scrapeSeatsSession.jar))),`.

**Comment by stash** [comment]
This has been fixed along with #59
